### PR TITLE
FBISCC-37: create Notify Integration

### PR DESCRIPTION
**Pull request by AlexFranklinHODDaT**
## What
Adds Notify integration to send the user's form inputs to an email address.

## Why
So that the user's query can be received and actioned by the SRC.

## How
* Adds utils method for sending an email with notify (reusable for future feedback and confirmation emails)
* Adds config file pulling notify api key, SRC casework email, and query email template from env vars (expandable with future templates and addresses)
* For the template ID, see [the template on Notify](https://www.notifications.service.gov.uk/services/7c8d0248-51b8-4795-b920-3ff84efb7faf/templates/837dc8ac-6abf-4f6a-9f0c-57a28ea7f43c)
* For the test api key, contact a team member to securely receive it
* For the test casework email, use the email address for FBIS CCF Developers on the [Notify team](https://www.notifications.service.gov.uk/services/7c8d0248-51b8-4795-b920-3ff84efb7faf/users)
* Adds 'submit' behaviour to send email to SRC casework email address including values inputted into the form by the user

## Test
Small refactor of existing tests to divide them into different directories. In this way we can separate different types of modules and possibly different categories of test in future.

Unit tests added for the new behaviour and utils method. [Proxyquire](https://www.npmjs.com/package/proxyquire) is used for dependency mocking where needed.

## Screenshots
Screenshot 2020-11-05 at 17 39 09

### CODE REVIEW

**Comment by sulthan-ahmed** [comment]
Love the PR 👍

File: apps/fbis-ccf/behaviours/submit.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const config = require('../../../config');
+const sendEmail = require('../../../lib/utils').sendEmail;
```
Comment (sulthan-ahmed) [inline apps/fbis-ccf/behaviours/submit.js:4]: made it reusable, saweeeeet sugaaaar

Alt Text

File: apps/fbis-ccf/behaviours/submit.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const fields = require('../fields/index');
+const uuidv4 = require('uuid').v4;
+
+module.exports = superclass => class Submit extends superclass {
+
+  saveValues(req, res, next) {
+    let emailData = Object.keys(fields).reduce((data, field) => {
```
Comment (sulthan-ahmed) [inline apps/fbis-ccf/behaviours/submit.js:11]: nice

File: apps/fbis-ccf/behaviours/submit.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      data[field] = req.form.historicalValues[field] || 'n/a';
+      return data;
+    }, {});
```
Comment (sulthan-ahmed) [inline apps/fbis-ccf/behaviours/submit.js:14]: learner question, how does the historicalValues work, is this the thing that was added by Andy Moody to HOF?

Comment (AlexFranklinHODDaT) [reply apps/fbis-ccf/behaviours/submit.js:14]: Yeah, Andy added it. For whatever reason, `values` is empty at this stage but `historicalValues` contains all form fields and anything else that's been added to the session.

**Review by sulthan-ahmed — APPROVED** [review]
great stuff, as you suggested it would be good to have a readme and as agreed we need to think about a scenario where we don't have an api key what should we do?

**Comment by sulthan-ahmed** [comment]
nice

**Pull request merged**